tests/ui/traits/struct-negative-sync-impl.rs

```diff
@@ -1,14 +1,21 @@
+//! Test negative Sync implementation on structs.
+//!
+//! Uses the unstable `negative_impls` feature to explicitly opt-out of Sync.
+
 #![feature(negative_impls)]
 
 use std::marker::Sync;
 
-struct Foo { a: isize }
-impl !Sync for Foo {}
+struct NotSyncStruct {
+    value: isize,
+}
+
+impl !Sync for NotSyncStruct {}
```

Comment:
`NotSyncStruct` -> `NotSync`, we avoid [Hungarian Notation](https://en.wikipedia.org/wiki/Hungarian_notation) in Rust